### DISCUSSION TITLE
Custom field client extension not working with select

### DISCUSSION THREAD

**Discussion by rijk** [prisma/prisma#21676] [Q&A] [answered]
### Bug description

I am not sure if this is a bug or working as designed; the docs don't mention this case.

Following the [example case](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result#add-a-custom-field-to-query-results):

```tsx
const prisma = new PrismaClient().$extends({
  result: {
    user: {
      fullName: {
        needs: { firstName: true, lastName: true },
        compute({ firstName, lastName }) {
          return `${firstName} ${lastName}`
        },
      },
    },
  },
})
```

### How to reproduce

Works as expected without a `select`:

```ts
const user = await prisma.user.findFirstOrThrow({ where: { email: 'test@me.com' }})
user.fullName // fullName: string
```

But when adding `select`:

```ts
const user = await prisma.user.findFirstOrThrow({
  where: { email: 'test@me.com' },
  select: { firstName: true, lastName: true },
})
```

The field is missing from the `user` object, even though its `needs` are present:

<img width="542" alt="image" src="https://github.com/prisma/prisma/assets/159500/f9484a45-7dfa-48f8-a8b6-cff44e5c6a39">

### Expected behavior

When `select` contains the fields defined in `needs`, the computed property should be available in the result object.

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 5.5.2
@prisma/client          : 5.5.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.5.1-1.aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Default Engines Hash    : aebc046ce8b88ebbcb45efe31cbe7d06fd6abc0a
Studio                  : 0.494.0
```

**Comment by SevInf** [comment, marked as answer]
This works as expected. Computed fields get their own entry in `select`:

```ts
const user = await prisma.user.findFirstOrThrow({
  where: { email: 'test@me.com' },
  select: { fullName: true },
})
```

Motivation for this design is that `needs` are implementation detail of particular field:  otherwise, if tomorrow `middleName` would also become a part of `fullName`, you'd have to update every single `select` that needs `fullName`.

**Reply by rijk** [reply]
Aaah, makes sense. Maybe it's good to add this to the [docs page](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result#add-a-custom-field-to-query-results)?

```ts
const user = await prisma.user.findFirst({select: { fullName: true }})
```
